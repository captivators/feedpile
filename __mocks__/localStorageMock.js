const localStorageMock = (() => {
  let store = {
    "profile": '{"name":"faiz"}',
    id_token: ""
  };
  return {
    getItem(key) {
      return store[key]
    },
    setItem(key, value) {
      store[key] = value.toString()
    },
    clear() {
      store = {}
    }
  };
})();

global.localStorage = localStorageMock;
