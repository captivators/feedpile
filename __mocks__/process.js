const process = {
  "env" : {
    "AUTH_ID":"test",
    "AUTH_DOMAIN":"user_id"
  }
}
global.process = process;
