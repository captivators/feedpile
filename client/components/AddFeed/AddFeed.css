.url-input {
  margin-bottom: 30px;
}

.radio-buttons {

}
