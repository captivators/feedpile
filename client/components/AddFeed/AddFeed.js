import React from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import {toggleModal, addFeed, setAddFeedCategoryId, setAddFeedUrl, setDisplayProgress} from '../../actions';
import {RadioButton, RadioButtonGroup} from 'material-ui/RadioButton';
import {connect} from 'react-redux';
import './AddFeed.css';

const styles = {
  block: {
    maxWidth: 250,
  },
  radioButton: {
    marginBottom: 16,
  },
};

/**
 * Dialogs can be nested. This example opens a Date Picker from within a Dialog.
 */
const AddFeed = (props) => {
  var userId = JSON.parse(localStorage.getItem('profile')).identities[0].user_id;

  const actions = [
    <FlatButton
      label="Cancel"
      primary={true}
      onTouchTap={() => {
        props.toggleModal(false);
      }}
    />,
    <RaisedButton
      label="Submit"
      primary={true}
      onTouchTap={() => {
        props.addFeed(props.urlInput, userId, props.selectedCategory)
        props.setDisplayProgress(true)
        props.toggleModal(false);
      }}
    />
  ];

  return (
    <div>
      {/* <RaisedButton label="Dialog With Date Picker" onTouchTap={this.handleOpen} /> */}
      <Dialog
        title="Enter an RSS feed URL and associate a category"
        actions={actions}
        modal={false}
        open={props.modalOpen}
        contentStyle={{width: '75%', maxWidth: '600px'}}
        onRequestClose={() => {
          props.toggleModal(false);
        }}
      >
        <div className="url-input">
          <TextField
            hintText="http://coolnews.rss"
            fullWidth={true}
            onChange={(e) => {
              props.setAddFeedUrl(e.target.value)
            }}
          />
        </div>
        <div className="radio-buttons">
          <RadioButtonGroup name="categories" defaultSelected="Other">
            {props.categories.map((category, i) => {
              return (<RadioButton
                  key={i}
                  value={props.categories[i].name}
                  label={props.categories[i].name}
                  onTouchTap={() => {
                    props.setAddFeedCategoryId(props.categories[i]._id)
                  }}
                  style={styles.radioButton}
              />)
            })}
          </RadioButtonGroup>
        </div>
      </Dialog>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    modalOpen: state.modalOpen,
    categories: state.categories,
    urlInput: state.addFeedUrl,
    selectedCategory: state.addFeedCategoryId
  }
}

export const Unwrapped = AddFeed;
export default connect(mapStateToProps, {toggleModal, addFeed,
setAddFeedCategoryId, setAddFeedUrl, setDisplayProgress})(AddFeed);
