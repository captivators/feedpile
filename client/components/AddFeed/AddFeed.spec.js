import React from 'react';

import '../../../__mocks__/localStorageMock'

import {Unwrapped as UnwrappedAddFeed} from './AddFeed'
import { shallow} from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('AddFeed take a snapshot', () => {
  const component = shallow(<UnwrappedAddFeed categories={[]}/>);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

