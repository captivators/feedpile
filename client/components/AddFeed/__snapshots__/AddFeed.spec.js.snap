// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFeed take a snapshot 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          onTouchTap={[Function]}
          primary={true}
          secondary={false}
      />,
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Submit"
          labelPosition="after"
          onTouchTap={[Function]}
          primary={true}
          secondary={false}
      />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    contentStyle={
      Object {
        "maxWidth": "600px",
        "width": "75%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    repositionOnUpdate={true}
    title="Enter an RSS feed URL and associate a category"
  >
    <div
      className="url-input"
    >
      <TextField
        disabled={false}
        floatingLabelFixed={false}
        fullWidth={true}
        hintText="http://coolnews.rss"
        multiLine={false}
        onChange={[Function]}
        rows={1}
        type="text"
        underlineShow={true}
      />
    </div>
    <div
      className="radio-buttons"
    >
      <RadioButtonGroup
        defaultSelected="Other"
        name="categories"
        style={Object {}}
      />
    </div>
  </Dialog>
</div>
`;
